openedx-unsupported/configuration
# fix: change default mongodb replica set to null

Configuration Pull Request
---

Make sure that the following steps are done before merging:


## Description

This PR changes the default value of `EDXAPP_MONGO_REPLICA_SET` from empty string (`""`) to null.

This is required due to unpinnig of pymongo via [edx-platform#30569](https://github.com/openedx/edx-platform/pull/30569) 
As mentioned in that PR, `pymongo >= 3.11` only accepts `None` as replica set if standalone MongoDB instance is used.
Empty string is not accepted as a valid value for replica set.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

1. Spinup an instance using a standalone MongoDB instance without passing `EDXAPP_MONGO_REPLICA_SET`
2. Ensure the instance spins up correctly, and all migrations run correctly.

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Private Ref: [BB-6773](https://tasks.opencraft.com/browse/BB-6773)
